proposals/0118-partitioned-epoch-reward-distribution.md

```diff
@@ -152,54 +152,59 @@ be computed by applying the following formula to the hash:
 I = (M * stake_address_hash) / 2^64
 ```
 
-### `EpochRewards` Sysvar Account
+### Reward Distribution
+
+The reward distribution phase happens after the reward computation phase, ie.
+during the second block in the epoch as per the current design. It lasts for `M`
+blocks (see previous section).
+
+#### `EpochRewards` Sysvar Account
 
-`EpochRewards` sysvar account records the total rewards and the amount of
-distributed rewards for the current epoch internally. And the account balance
-reflects the amount of pending rewards to distribute.
+The `EpochRewards` sysvar account records whether the rewards distribution phase
+is in progress, as well as the details needed to resume distribution when
+starting from a snapshot during the reward distribution phase. These details
+include: the parameters needed to recalculate the reward partitions, the total
+rewards to be distributed, and the amount of rewards distributed so far.
 
 The layout of `EpochRewards` sysvar is shown in the following pseudo code.
 
 ```
-struct RewardRewards{
-   // total rewards for the current epoch, in lamports
-   total_rewards: u64,
+struct EpochRewards{
+   // whether rewards distribution is active
```

Comment:
Is it also true during rewards calculation?

Reply:
Technically (at least in the current implementation), rewards calculation completes before the sysvar is populated, although both steps happen in the same block and both complete before any transactions are processed.

I think the existing comment is correct, but I am happy to make more explicit if there's additional information that would be helpful here.

Reply:
Ok makes sense. So is it set to active before the first block of an epoch is processed?

Reply:
> So is it set to active before the first block of an epoch is processed?

Yep, correct

Reply:
Great and maybe worth one more small clarification that if the last partition of rewards is distributed at the beginning of slot X, it will longer be active in slot X when processing transactions. Just want to avoid an off-by-one issue with other implementations.

Reply:
Added some clarification. Let me know if it reads okay.